backend/app/modules/uq/service/preliminary.py

```diff
@@ -0,0 +1,49 @@
+from datetime import datetime
+from typing import Protocol, Optional, Tuple, runtime_checkable
+
+from app.context.manager import Context
+from app.infrastructure import database
+from app.modules.uq.core import calculate_uq, Scores
+from app.pydantic import Model
+
+
+@runtime_checkable
+class Antisybil(Protocol):
+    def get_antisybil_status(
+        self, _: Context, user_address: str
+    ) -> Optional[Tuple[float, datetime]]:
+        ...
+
+
+@runtime_checkable
+class Epoch0Poap(Protocol):
+    def has_poap(self, context: Context, address: str) -> bool:
+        ...
+
+
+@runtime_checkable
+class IdentityPoap(Protocol):
+    def has_poap(self, context: Context, address: str) -> bool:
```

Comment:
Most certainly we will not issue a poap after the identity calls, so it will be just added to the whitelist.

Reply:
changed